services/QuillLMS/client/app/bundles/Teacher/components/salesForm/index.tsx

```diff
@@ -0,0 +1,174 @@
+    } else {
+      setSelectedSchool(value);
+    }
+  };
+
+  function handleDistrictSearchChange(value: string) {
+    if(value === DISTRICT_NOT_LISTED) {
+      setDistrictNotListed(true);
+      setSelectedDistrict('');
+    } else {
+      setSelectedDistrict(value);
+    }
+  };
+
+  function handleFormSubmission(e: React.MouseEvent) {
+    e.preventDefault();
+    const salesFormSubmission = {
+      first_name: firstName,
+      last_name: lastName,
+      email: email,
+      phone_number: phoneNumber,
+      zipcode: zipcode,
+      collection_type: schoolOrDistrict.toLowerCase(),
+      school_name: selectedSchool,
+      district_name: selectedDistrict,
+      school_premium_count_estimate: parseInt(schoolPremimumEstimate),
+      teacher_premium_count_estimate: parseInt(teacherPremimumEstimate),
+      student_premium_count_estimate: parseInt(studentPremimumEstimate),
+      submission_type: type,
+      comment: comments
+    }
+    const formErrors = validateSalesForm(salesFormSubmission);
+    if(Object.keys(formErrors).length) {
+      setErrors(formErrors)
+    } else {
+      setErrors({});
+      submitSalesForm(salesFormSubmission).then(response => {
+        if(response.error) {
+          const submissionError = { [SUBMISSION_ERROR]: response.error };
+          setErrors(submissionError)
+        } else {
+          setFormSubmitted(true);
+        }
+      });
+    }
+  }
+
+  return(
+    <div className="sales-form-container">
+      {!formSubmitted && <form className="container">
+        <UpperFormFields
+          email={email}
+          errors={errors}
+          firstName={firstName}
+          handleUpdateField={handleUpdateField}
+          lastName={lastName}
+          phoneNumber={phoneNumber}
+          type={type}
+          zipcode={zipcode}
+        />
+        <SchoolAndDistrictFields
+          districtIsSelected={districtIsSelected}
+          districtNotListed={districtNotListed}
+          districts={districts}
+          errors={errors}
+          handleDistrictSearchChange={handleDistrictSearchChange}
+          handleSchoolSearchChange={handleSchoolSearchChange}
+          handleUpdateField={handleUpdateField}
+          schoolIsSelected={schoolIsSelected}
+          schoolNotListed={schoolNotListed}
+          schools={schools}
+          selectedDistrict={selectedDistrict}
+          selectedSchool={selectedSchool}
+        />
+        <LowerFormFields
+          comments={comments}
+          errors={errors}
+          handleUpdateField={handleUpdateField}
+          schoolPremimumEstimate={schoolPremimumEstimate}
+          studentPremimumEstimate={studentPremimumEstimate}
+          teacherPremimumEstimate={teacherPremimumEstimate}
+        />
+        {errors[SUBMISSION_ERROR] && <p className="error-text">{errors[SUBMISSION_ERROR]}</p>}
+        <button className="submit-button quill-button contained primary medium" onClick={handleFormSubmission}>Submit</button>
```

Comment:
Let's add a `focus-on-light` class here to help make this accessible.

Reply:
added!